### PR TITLE
Ats frontend 

### PR DESCRIPTION
**Pull request by github4touchdouble**
limited iframe support only

### CODE REVIEW

File: app/[lang]/(board)/board/[slug]/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  excludeFooter?: boolean
+}
+
+export default function BoardLayout({ children, params: { lang } }: boardLayoutProps) {
+    const [user, setUser] = useState<User | null>(null)
+    const [isLoading, setIsLoading] = useState(true)
+    const [dict, setDict] = useState<Record<string, any> | null>(null)
+    const router = useRouter()
+    const searchParams = useSearchParams()
+    const refreshToken = searchParams.get("refresh_token")
```
Comment (carlobortolan) [inline app/[lang]/(board)/board/[slug]/layout.tsx:38]: We don't need refresh_token, noredirect, mode, eType, etc.
Those are query params used for managing redirects when applying with apply.embloy.com, but not needed for the job board

File: app/[lang]/(board)/board/[slug]/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    const excludeHeader = searchParams.get("exclude_header") === "true"
+    const excludeFooter = searchParams.get("exclude_footer") === "true"
+
+    useEffect(() => {
+      const fetchDictionaryAndUser = async () => {
+        const dictionary = await getDictionary(lang)
+        setDict(dictionary)
+  
+        setIsLoading(true)
+        const { response, err } = await getCurrentUser(refreshToken ?? undefined)
```
Comment (carlobortolan) [inline app/[lang]/(board)/board/[slug]/layout.tsx:51]: `getCurrentUser()` should work as well.

File: next.config.mjs [hunk 3]
```diff
@@ -7,7 +7,7 @@ import "./env.mjs"
 const nextConfig = {
   reactStrictMode: true,
   images: {
-    domains: ["avatars.githubusercontent.com", "f005.backblazeb2.com", "embloy2.blob.core.windows.net", "embloy.s3.eu-central-1.amazonaws.com"],
+    domains: ["avatars.githubusercontent.com", "f005.backblazeb2.com", "embloy2.blob.core.windows.net", "embloy.s3.eu-central-1.amazonaws.com", "api.embloy.com"],
```
Comment (carlobortolan) [inline next.config.mjs:10]: Do we need this?
All images should be stored on embloy.s3

File: app/[lang]/(board)/board/[slug]/[job_slug]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,488 @@
+"use client";
+import { getDictionary } from "@/app/[lang]/dictionaries";
+import { getListedJob, getListedJobs } from "@/lib/api/jobs";
+import { Job } from "@/types/job-schema";
+import { useRouter } from "next/navigation";
+import { useEffect, useRef, useState } from "react";
+import { FaAt, FaFacebook, FaGithub, FaInstagram, FaLink, FaLinkedin, FaPhone, FaTwitter } from "react-icons/fa";
```
Comment (carlobortolan) [inline app/[lang]/(board)/board/[slug]/[job_slug]/page.tsx:7]: Use [lucide.dev](https://lucide.dev) icons instead, just as you did in line 10 with: 
```tsx
import { AlignEndHorizontal, ArrowLeft, Bookmark, MapPin, Share2 } from "lucide-react";
```
(If some icons from [lucide.dev](https://lucide.dev) are missing / not found in ts, you might need to wait for #22 to be merged first.)

**Review by carlobortolan — CHANGES_REQUESTED** [review]
Added some code-style comments; just two things (+1 UI suggestion):
1. Make sure that the project builds correctly 
2. When opening the company page without being logged in, everything works. However, when I first log in and then open a board (e.g., http://localhost:3000/en-US/board/3), nothing is shown / the page doesn't load correctly:

https://github.com/user-attachments/assets/31898772-28d4-420b-8f94-5d81b8b066ed

3. Some UI suggestions:
 - Add search bars on top of the list on md and lg screens (otherwise, the search bars might be hidden when embedding the job board on other sites)
 - Make the job titles clickable & add some hover effect
 - Remove the learn more button/link on mobile; 
 - Maybe highlight the apply button using the embloy accent color (?)

File: next.config.mjs (hunk 3 above)
Comment (github4touchdouble) [reply next.config.mjs:10]: ich nutze nur das was in dem json von der api steht lol

Comment (carlobortolan) [reply next.config.mjs:10]: Ja aber eigentlich served api.embloy.com keine dateien, oder haben wir etwa user mit image attachments mit api.embloy.com als host?

Comment (github4touchdouble) [reply next.config.mjs:10]: doch. hat mich ultra gewndert

Comment (carlobortolan) [reply next.config.mjs:10]: dann schick mir mal bitte die user ids

File: app/[lang]/(board)/board/[slug]/[job_slug]/page.tsx (hunk 4 above)
Comment (github4touchdouble) [reply app/[lang]/(board)/board/[slug]/[job_slug]/page.tsx:7]: transfered into separate isuue

**Comment by github4touchdouble** [comment]
guru tech panda sir, pls review

**Review by carlobortolan — APPROVED** [review]
Dear sir bro, this is very :spaghetti: code which didn't build, run or work on my machine (and on Vercel).
I fixed it (see https://github.com/Embloy/Embloy-Core-Client/pull/24/commits/77104937f6026b06d97fa79409435236784bb31a) so that it now uses the correct TS types and doesn't crash when you open a job board for a non-existing company and updated the apply link so that you can now apply correctly. I also updated it so that it shows a "not found" message for companies without active jobs (see https://github.com/Embloy/Embloy-Core-Client/pull/24/commits/55b82204043574dae3c17a88665dcb91a6d80ed3).

Next time, please:
1. **Build your code locally before committing** it with `pnpm run build`
2. **Test the basic functionality** (e.g., applying for a job)
3. **Periodically fetch and pull the newest changes** with `git pull origin main` (no idea how you managed to make it work on your machine using the old deprecated job types)

Besides that, everything looks great! (I really really like this UI btw)

Thanking you kindly,
Sir Sir Sir Andrea Tatti Sir

**Pull request merged**